java/src/main/java/org/whispersystems/signalservice/loki/api/LokiAPI.kt

```diff
@@ -250,8 +250,12 @@ class LokiAPI private constructor(private val userHexEncodedPublicKey: String, p
     private fun updateLastMessageHashValueIfPossible(target: LokiAPITarget, rawMessages: List<*>) {
         val lastMessageAsJSON = rawMessages.lastOrNull() as? Map<*, *>
         val hashValue = lastMessageAsJSON?.get("hash") as? String
+        val expiration = lastMessageAsJSON?.get("expiration") as? Int
         if (hashValue != null) {
             database.setLastMessageHashValue(target, hashValue)
+            if (expiration != null) {
+                LokiPushNotificationAcknowledgement.acknowledgeDeliveryForMessageWith(hashValue, expiration, userHexEncodedPublicKey)
```

Comment:
Instead of doing this here let's add it to `LokiAPI.getMessages()` and `LokiPoller.poll(target:deferred:)`. Right now it's an unexpected side effect of calling `updateLastMessageHashValueIfPossible(target:rawMessages:)` as you would expect that to just update the last message hash value.